### PR TITLE
Wrap Selections in both Quicktags Editor and TinyMCE w/ Shortcodes

### PR DESCRIPTION
**Pull request by JayWood**
Speaks for itself, this will wrap the user's selection in the short code tag, complete w/ closing tags.

### CODE REVIEW

File: js/shortcode-quicktag-button.js [hunk 1]
```diff
@@ -111,6 +113,24 @@ window.wp_sc_buttons.qt = (function(window, document, $, QTags, buttons, scbutto
...
+			 */
+			if ( mce_selection ) {
+				shortcode += mce_selection + close_tag;
+			} else if ( ! btn.isVisual ) {
+				var canvas = document.getElementById( 'content' ),
+					text = canvas.value,
+					startPos = canvas.selectionStart,
+					endPos = canvas.selectionEnd;
+
+				window.console.log( startPos,endPos );
```
Comment (jtsternberg) [inline js/shortcode-quicktag-button.js]: is this log necessary? Also, if you're logging, use the logging method, `scbuttons.log`.

File: js/shortcode-quicktag-button.js [hunk 2]
```diff
@@ -103,7 +103,9 @@ window.wp_sc_buttons.qt = (function(window, document, $, QTags, buttons, scbutto
 
 		btn.buildShortCode = function( shortcode_params ) {
 
-			var shortcode = '['+ params.slug;
+			var shortcode = '['+ params.slug,
+				close_tag = '[/' + params.slug + ']',
+				mce_selection = ( tinymce.activeEditor ) ? tinymce.activeEditor.selection.getContent() : false;
```
Comment (jtsternberg) [inline js/shortcode-quicktag-button.js:108]: Please, no inline var declaration. One var statement/line per var declaration. ([reference](http://benalman.com/news/2012/05/multiple-var-statements-javascript/))

File: js/shortcode-quicktag-button.js [hunk 3]
```diff
@@ -103,7 +103,9 @@ window.wp_sc_buttons.qt = (function(window, document, $, QTags, buttons, scbutto
...
```
Comment (jtsternberg) [inline js/shortcode-quicktag-button.js]: Does this work on/in the text view? I know there's a way to get the selection in the text view as well.

**Comment by jtsternberg** [comment]
I prefer that the close tag not be used if there is no selection. (i.e. use the self-closing shortcode)

**Comment by jtsternberg** [comment]
This should be a parameter that is registered when `_Shortcode_Button_` is first initiated, whether shortcode can wrap existing text. if not, should use default (existing) handling.

**Comment by JayWood** [comment]
The log was just an oopsie on my part, but i'm sad that inline var declarations aren't a thing, they look so much cleaner.

I'll see what I can do about the registration parameter for closing tags.  But that begs the question, if the user says 'yes' to closing tags, ie, the shortcode must use a selection, then wouldn't we require the user to make a selection before they can even use the shortcode?

**Comment by JayWood** [comment]
Oh, and yes this currently works with the text only mode as well.

**Comment by jtsternberg** [comment]
By 'user', I think you mean the dev who initiates, right? And no, i wasn't saying a selection should be required, though the parameter could take 'required' as a value if you want to make your PR work that way. the parameter would default to false (not allowed), could toggle to true (allowed) or 'required'.

**Comment by jtsternberg** [comment]
Thanks dude!

**Pull request merged**